gr-blocks/lib/rotator_cc_impl.cc

```diff
@@ -63,7 +102,38 @@ int rotator_cc_impl::work(int noutput_items,
       for (int i=0; i<noutput_items; i++)
       	out[i] = d_r.rotate(in[i]);
 #else
-    d_r.rotateN(out, in, noutput_items);
+    int items_before_update, items_after_update;
+    /* If there is a phase increment update scheduled, handle rotation in two
+     * steps and update the phase increment in between. */
+    const uint64_t n_written = nitems_written(0);
+    if (d_inc_update_pending &&
+        d_idx_next_inc_update > n_written &&
+        d_idx_next_inc_update <= (n_written + noutput_items)) {
+
+        items_before_update = d_idx_next_inc_update - n_written - 1;
```

Comment:
I've pushed this fix also. The `-1` is needed here in order for the new phase increment to start taking effect on sample index `d_idx_next_inc_update`.  I realised this after introducing the QA tests.